src/gui/connectionvalidator.cpp

```diff
@@ -223,18 +245,18 @@ void ConnectionValidator::slotAuthFailed(QNetworkReply *reply)
 void ConnectionValidator::slotAuthSuccess()
 {
     _errors.clear();
-    if (!_updateConfig) {
-        reportResult(Connected);
+    if (_mode != ConnectionValidator::ValidationMode::ValidateAuth) {
```

Comment:
Maybe `Q_ASSERT` you're not in `ValidateServer`? I assume the use of `!=` rather than `>` implies this.